### PR TITLE
Pkg: h5py

### PR DESCRIPTION
**Pull request by ax3l**
### Description

Add the h5py package.
X-ref: #1799

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

File: packages/libhdf5/meta.yaml [hunk 1]
```diff
@@ -20,8 +20,8 @@ build:
         -DCMAKE_INSTALL_PREFIX=hdf5 \
         -DH5_HAVE_GETPWUID=0 \
         -DH5_HAVE_SIGNAL=0 \
-        -DBUILD_SHARED_LIBS=0 \
-        -DBUILD_STATIC_LIBS=1 \
+        -DBUILD_SHARED_LIBS=1 \
```
Comment (bmaranville) [inline packages/libhdf5/meta.yaml:23]: Building shared libs is not strictly necessary, I think.  After looking over the build system in h5py, it looks like they are loading the .so file to do 3 things, and then discarding it...

1. extract version number (major, minor, release)
2. test for ```mpi``` support
3. test for ```ros3``` support

**Comment by bmaranville** [comment]
I added these patches:
```packages/h5py/patches/configure.patch```:
```patch
diff --git a/setup_configure.py b/setup_configure.py
index 16c355b..2273ae3 100644
--- a/setup_configure.py
+++ b/setup_configure.py
@@ -183,7 +183,8 @@ class BuildConfig:
 class HDF5LibWrapper:
 
     def __init__(self, libdirs):
-        self._load_hdf5_lib(libdirs)
+        # self._load_hdf5_lib(libdirs)
+        pass
 
     def _load_hdf5_lib(self, libdirs):
         """
@@ -248,20 +249,21 @@ class HDF5LibWrapper:
 
         Raises an exception if anything goes wrong.
         """
-        import ctypes
-        from ctypes import byref
+        # import ctypes
+        # from ctypes import byref
 
-        major = ctypes.c_uint()
-        minor = ctypes.c_uint()
-        release = ctypes.c_uint()
+        # major = ctypes.c_uint()
+        # minor = ctypes.c_uint()
+        # release = ctypes.c_uint()
 
-        try:
-            self._lib.H5get_libversion(byref(major), byref(minor), byref(release))
-        except Exception:
-            print("error: Unable to find HDF5 version")
-            raise
+        # try:
+        #     self._lib.H5get_libversion(byref(major), byref(minor), byref(release))
+        # except Exception:
+        #     print("error: Unable to find HDF5 version")
+        #     raise
 
-        return int(major.value), int(minor.value), int(release.value)
+        # return int(major.value), int(minor.value), int(release.value)
+        return (1, 12, 1)
 
     def load_function(self, func_name):
         try:
@@ -277,7 +279,7 @@ class HDF5LibWrapper:
         return True
 
     def has_mpi_support(self):
-        return self.has_functions("H5Pget_fapl_mpio", "H5Pset_fapl_mpio")
+        return False
 
     def has_ros3_support(self):
-        return self.has_functions("H5Pget_fapl_ros3", "H5Pset_fapl_ros3")
+        return False
```
and ```packages/h5py/patches/build.patch```:
```patch
diff --git a/setup_build.py b/setup_build.py
index 6b66789..448774a 100644
--- a/setup_build.py
+++ b/setup_build.py
@@ -104,7 +104,7 @@ class h5py_build_ext(build_ext):
             settings['include_dirs'] += [mpi4py.get_include()]
 
         # TODO: should this only be done on UNIX?
-        if os.name != 'nt':
+        if False:
             settings['runtime_library_dirs'] = settings['library_dirs']
 
         def make_extension(module):
```
and then modified ```packages/h5py/meta.yaml```:
```patch
diff --git a/packages/h5py/meta.yaml b/packages/h5py/meta.yaml
index 3850c3d..82d09b4 100644
--- a/packages/h5py/meta.yaml
+++ b/packages/h5py/meta.yaml
@@ -5,6 +5,9 @@ package:
 source:
   url: https://github.com/h5py/h5py/archive/3.6.0.tar.gz
   sha256: 840fca0787bca8ad2729b9c20e06a16f219e7a24ccc59fdf5177a05ec8472b2d
+  patches:
+    - patches/configure.patch
+    - patches/build.patch
 
 requirements:
   run:
```

**Comment by bmaranville** [comment]
After that I was able to build h5py, and it seems to work:
image

**Comment by hoodmane** [comment]
@bmaranville could you open a pr on top of this one with those changes?

**Comment by bmaranville** [comment]
I created a branch based on [ax3l](https://github.com/ax3l/pyodide/tree/pkg-h5py) and pushed it to https://github.com/bmaranville/pyodide/tree/pkg-h5py but for some reason it won't let me create a pull request on top of this one.  In github the chooser for "base repository" doesn't include an entry for ax3l/pkg-h5py
image

**Comment by rth** [comment]
Just a separate PR to pyodide/pyodide:main is fine. Once the CI passes there we would be able to merge libhdf5 from this PR followed by your PR. Thanks!

**Comment by bmaranville** [comment]
I created #2411 based on this PR, but rebasing to the current main, and with the fixes described above.

**Comment by ax3l** [comment]
Thanks for taking this on @bmaranville :+1:

**Comment by rth** [comment]
Resolved in https://github.com/pyodide/pyodide/pull/2411

Thanks for working on this!

**Pull request closed without merging**